app/assets/javascripts/modules/thumb_slider.js

```diff
@@ -0,0 +1,97 @@
+(function( global ) {
+  'use strict';
+  var Module = (function() {
+    var selector;
+
+    var containerTemplate =
+      jQuery(
+        '<div class="sul-embed-thumb-slider-container"></div>'
+      );
+
+    var toggleControl =
+      jQuery(
+        '<div class="sul-i-navigation-show-more-1 sul-embed-thumb-slider-open-close"' +
+          ' aria-expanded="true" aria-label="toggle thumbnail viewer">' +
+        '</div>'
+      );
+
+    var thumbList =
+      jQuery(
+        '<div class="sul-embed-thumb-slider" style="display:none">' +
+          '<ul></ul>' +
+        '</div>'
+      );
+
+    function hideInitialMediaObjects() {
+      $('[data-slider-object]').each(function(index, object) {
+        if(index !== 0) {
+          $(object).hide();
+        }
+      });
+    }
+
+    function hidePanelIfNotMultiple(thumbnails) {
+      if(thumbnails.length < 2 && thumbList.find('li').length === 0) {
+        containerTemplate.hide();
+        return;
+      }
+    }
+
+    function addToggleControlBehavior() {
+      toggleControl.on('click', function() {
+        thumbList.slideToggle();
+      });
+    }
+
+    function appendSliderToFrame(frame) {
+      var container = containerTemplate.append(toggleControl, thumbList);
+      frame.append(container);
+    }
+
+    return {
+      init: function(providedSelector, options) {
+        this.options = options;
+        selector = providedSelector;
+        appendSliderToFrame(this.scrollFrame());
+        hideInitialMediaObjects();
```

Comment:
My concern with naming it based on the initial implementation (i.e. `MediaThumbSlider`) is that it will never be used outside of the context of the initial implementation.

It is written to be re-usable in other contexts, so keeping the initial implementation out of the name seems prudent.